File: leetcode-04/solutions/python3/0495.py
class Solution:
    def findPoisonedDuration(self, timeSeries: List[int], duration: int) -> int:
        if not timeSeries or duration == 0:
            return 0

        ans = 0

        for i in range(len(timeSeries) - 1):
            ans += min(timeSeries[i + 1] - timeSeries[i], duration)

        return ans + duration

File: leetcode-04/solutions/cpp/0495.cpp
class Solution {
 public:
  int findPoisonedDuration(vector<int>& timeSeries, int duration) {
    if (timeSeries.empty() || duration == 0) return 0;

    int ans = 0;

    for (int i = 0; i < timeSeries.size() - 1; i++)
      ans += min(timeSeries[i + 1] - timeSeries[i], duration);

    return ans + duration;
  }
};